src/librbd/internal.cc

```diff
@@ -113,6 +121,18 @@ int invoke_async_request(ImageCtx *ictx, const std::string& request_type,
         }
 
         r = remote_request();
+        if (r == -EOPNOTSUPP && request_lock) {
+          AioCompletion *comp = aio_create_completion();
+          ictx->image_watcher->request_lock(boost::bind(&complete_request, _1),
```

Comment:
Except one issue -- noise in the log:

2016-11-16 15:27:20.087375 7f02db7fe700 -1 librbd::AioCompletion: completed invalid aio_type: 4

@dillaman May I update AioCompletion::complete not to log error for AIO_TYPE_NONE?